### ISSUE TITLE
plone.formwidget.recurrence i18n.py poorly internationalized

### ISSUE THREAD

**Issue by vincentfretin** [plone/Products.CMFPlone#2897]
plone.formwidget.recurrence/plone/formwidget/recurrence/browser/i18n.py is poorly internationalized.

In the plone.pot file we have 
```
#: plone.formwidget.recurrence/plone/formwidget/recurrence/browser/i18n.py:31
msgid "monthly_day_of_month_4"
msgstr ""
```

It seems this i18n.py is poorly internationalized. The default message should be in the code, but it seems the default message in only in the `en/LC_MESSAGES/plone.po` file. This should be fixed in i18n.py.
There is not even a translation in English for this msgid...

```
#: plone.formwidget.recurrence/plone/formwidget/recurrence/browser/i18n.py:128
msgid "no_end_after_n_occurrences"
msgstr ""
```

In `en/LC_MESSAGES/plone.po` we have the following message
"Error: The \"After N occurrences\"-field must be between 1 and 1000"
But again, this should be in i18n.py.
If other language doesn't have a translation for this, the frontend will show "no_end_after_n_occurrences" and not the English default message, because the default message is not in the code.

```
#: plone.formwidget.recurrence/plone/formwidget/recurrence/browser/i18n.py:124
msgid "no_repeat_every"
msgstr ""
```
=> "Error: The \"Repeat every\"-field must be between 1 and 1000"

```
#: plone.formwidget.recurrence/plone/formwidget/recurrence/browser/i18n.py:126
msgid "no_repeat_on"
msgstr ""
```
=> "Error: \"Repeat on\"-value must be selected"

```
#: plone.formwidget.recurrence/plone/formwidget/recurrence/browser/i18n.py:120
msgid "no_template_match"
msgstr ""
```
=> "No matching recurrence template"

```
#: plone.formwidget.recurrence/plone/formwidget/recurrence/browser/i18n.py:119
msgid "unsupported_features"
msgstr ""
```
=> "Warning: This event uses recurrence features not supported by this widget. Saving the recurrence may change the recurrence in unintended ways: "

**Comment by erral** [comment]
Indeed. This was also discussed last year on the community forum: https://community.plone.org/t/plone-5-1-translations/6006/32 and there is even a ticket regarding this: https://github.com/plone/plone.formwidget.recurrence/issues/24

**Comment by erral** [comment]
At the moment the best way to translate the recurrence widget, is to install a coredev.buildout based Plone site, translate the PO files and check whether the translations are good.

**Comment by jensens** [comment]
Verified still present. All message IDs in i18n.py use bare msgids without default= kwarg. Untranslated locales show raw IDs like 'no_end_after_n_occurrences'.
